cli/src/spend_utils.rs

```diff
@@ -96,15 +105,37 @@ where
         )?;
         Ok((message, spend))
     } else {
-        let from_balance = rpc_client
+        let mut from_balance = rpc_client
             .get_balance_with_commitment(from_pubkey, commitment)?
             .value;
-        let from_rent_exempt_minimum = if amount == SpendAmount::RentExempt {
-            let data = rpc_client.get_account_data(from_pubkey)?;
-            rpc_client.get_minimum_balance_for_rent_exemption(data.len())?
-        } else {
-            0
-        };
+        let from_rent_exempt_minimum =
+            if amount == SpendAmount::RentExempt || amount == SpendAmount::Available {
+                let data = rpc_client.get_account_data(from_pubkey)?;
+                rpc_client.get_minimum_balance_for_rent_exemption(data.len())?
+            } else {
+                0
+            };
+        if amount == SpendAmount::Available {
+            if let Some(account) = rpc_client
+                .get_account_with_commitment(from_pubkey, commitment)?
+                .value
+            {
+                if account.owner == solana_sdk::stake::program::id() {
+                    let state = stake::get_account_stake_state(
+                        rpc_client,
+                        from_pubkey,
+                        account,
+                        true,
+                        None,
+                        false,
+                    )?;
+                    if let Some(active_stake) = state.active_stake {
+                        from_balance = from_balance.saturating_sub(active_stake);
```

Comment:
Unfortunately, there's one more way this is broken.
Currently, the stake program takes a conservative approach to partially activated accounts.
https://github.com/anza-xyz/agave/blob/025ed451c597df8abb7e271f1ec80cf4998dd0fb/programs/stake/src/stake_state.rs#L742-L745

In other words, if a stake account has some active stake, the entire delegation at the point when it started delegating is locked to withdrawals, even if some of it is still activating over multiple epochs.

The fix here is to also subtract `state.activating_stake.unwrap_or_default()`

Sorry for not catching this sooner; in truth, we ought to work up a series of withdrawal [integration tests here](https://github.com/anza-xyz/agave/blob/master/cli/tests/stake.rs), using `SpendAmount::Available` (and `SpendAmount::All`) to catch all these edge cases. Please let me know if that's something you're interested/willing to work on.

Reply:
No problem !
So when I have `Some state.active_stake` I need to also subtract `activating_stake` at L134?

Regarding the stake account state I see other fields like `delegated_stake` `deactivating_stake`. Is it something to take into consideration?

About the integration tests, should it be done in this PR or in another one?

Reply:
> So when I have `Some state.active_stake` I need to also subtract `activating_stake` at L134?

No, when you have *any* activating_stake it needs to be subtracted. So this needs to happen outside of the `if let Some(active_stake)` case.

> Regarding the stake account state I see other fields like delegated_stake deactivating_stake. Is it something to take into consideration?

No, `deactivating_stake` is already included in `active_stake`, and `delegated_stake` doesn't reflect activation or deactivation history.
However, to understand all the fields better, I recommend you start with [this method](https://github.com/anza-xyz/agave/blob/baed522d2b9843aae257f63f4ef371dc6255e289/cli/src/stake.rs#L2322), and read upstream to `Delegation::stake_activating_and_deactivating()`

> About the integration tests, should it be done in this PR or in another one?

If you are up for it in this PR, that would be best. That way we can be confident we are not merging any edge-case bugs.

Thank you!